prusti-encoder/src/encoders/typ.rs

```diff
@@ -29,16 +35,21 @@ pub enum TypeEncoderOutputRefSub<'vir> {
 // TODO: should output refs actually be references to structs...?
 #[derive(Clone, Debug)]
 pub struct TypeEncoderOutputRef<'vir> {
-    pub snapshot_name: &'vir str,
-    pub predicate_name: &'vir str,
-    pub from_primitive: Option<&'vir str>,
-    pub to_primitive: Option<&'vir str>,
+    /// Constructs the Viper predicate application.
+    pub predicate_ref: PredicateIdent<'vir, UnaryArity<'vir>>,
     pub snapshot: vir::Type<'vir>,
-    pub function_unreachable: &'vir str,
-    pub function_snap: &'vir str,
+    /// Construct snapshot from snapshots of fields or for primitive types
+    /// from the single Viper primitive value. `None` for generic parameters.
```

Comment:
`Option` is not communicating intent very well here. Can these fields be put into `specifics`? Maybe a new variant is needed?

Reply:
Moved it to `specifics` as you suggest and indeed it's much cleaner.